### PR TITLE
Move verify-gendocs into Makefile and tighten Makefile safety

### PR DESCRIPTION
**Pull request by WillemMali**
I've moved `verify-gendocs` into the Makefile, so it can reuse the configuration values from the Makefile and be easier to inspect (through fewer layers of indirection, `make` instead of `make -> script`); I've also made it shorter and IMO a bit more readable. It also outputs a diff on failure now, so it's clearer what went wrong/what commands the failure applies to.

There's one part of the script I didn't understand so I left it with a FIXME (the script unsets `KOPS_STATE_STORE`).

Another change I made is setting the `-euC` flags on the Makefile, so we can hopefully let fewer errors slip through by accident.

The last change in the diff enforces the order of execution of a few targets which need to be ordered according to the preceding comment; this was not enforced before (it would've failed when executing `make` in parallel).

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kops/2617)

### CODE REVIEW

**Comment by chrislovecnm** [comment]
@k8s-bot ok to test

File: Makefile [hunk 1]
```diff
@@ -414,13 +421,28 @@ verify-packages:
 
 .PHONY: verify-gendocs
 verify-gendocs: kops
-	KOPS_STATE_STORE= hack/verify-gendocs.sh
+	@: FIXME: why unset this?; \
+	export KOPS_STATE_STORE=; \
```
Comment (chrislovecnm) [inline Makefile]: I do not think we need to anymore. You can test by removing and setting the car in your shell and seeing if running the command will embed the state store in the docs. I think this issue was already addressed in another pr.

**Comment by chrislovecnm** [comment]
@justinsb thoughts?

**Comment by WillemMali** [comment]
@chrislovecnm I followed your suggestion regarding the previous version of the PR and tested it.

I just noticed I removed the cleanup of the generated docs directory, I could re-add it or leave it as-is, because `mktemp`-made directories are cleaned up by your OS anyway (at restarts or at scheduled intervals).

**Comment by justinsb** [comment]
This LGTM but I'd like to hold merge until after 1.6.1 if that's OK (should be today / this weekend).

**Comment by chrislovecnm** [comment]
We need a rebase

**Pull request merged**